source/core/slang-string-slice-pool.h

```diff
@@ -0,0 +1,50 @@
+#ifndef SLANG_STRING_SLICE_POOL_H
+#define SLANG_STRING_SLICE_POOL_H
+
+#include "slang-string.h"
+
+#include "list.h"
+#include "slang-memory-arena.h"
+#include "dictionary.h"
+
+namespace Slang {
+
+class StringSlicePool
+{
+public:
+    /// Handle of 0 is null. If accessed will be returned as the empty string
+    enum class Handle : uint32_t;
+    typedef UnownedStringSlice Slice;
+
+        /// Returns the index of a slice, if contained, or -1 if not found
+    int findIndex(const Slice& slice) const;
+
+        /// True if has the slice
+    bool has(const Slice& slice) { return findIndex(slice) < 0; }
```

Comment:
Is this test backward? I would have assumed a negative index means "missing" not "present."

Reply:
yeah - good catch